External.CustomerReviewsModule/External.CustomerReviewsModule/External.CustomerReviewsModule.Data/Services/CustomerReviewService.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Linq;
+using External.CustomerReviewsModule.Core.Models;
+using External.CustomerReviewsModule.Core.Models.Search;
+using External.CustomerReviewsModule.Core.Services;
+using External.CustomerReviewsModule.Data.Converters;
+using External.CustomerReviewsModule.Data.Model;
+using External.CustomerReviewsModule.Data.Repositories;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Data.Infrastructure;
+
+namespace External.CustomerReviewsModule.Data.Services
+{
+    public class CustomerReviewService : ServiceBase, ICustomerReviewService
+    {
+        private readonly Func<ICustomerReviewRepository> _repositoryFactory;
+
+        public CustomerReviewService(Func<ICustomerReviewRepository> repositoryFactory)
+        {
+            _repositoryFactory = repositoryFactory;
+        }
+
+        public CustomerReview[] GetByIds(string[] ids)
+        {
+            using (var repository = _repositoryFactory())
+            {
+                return repository.GetByIds(ids).Select(x => x.ToModel(AbstractTypeFactory<CustomerReview>.TryCreateInstance())).ToArray();
+            }
+        }
+
+        public void SaveCustomerReviews(CustomerReview[] items)
```

Comment:
Single Save method is a common practice in VC, it has unified stable structure. Not sure on the purpose of this specific sample in details, but if it is best practice for VC custom module - I think we need to leave it here. The same for array parameters - better to have methods allowing bulk crud operations, otherwise we need to include them later to avoid lot of multiple queries on single object.

Reply:
@yecli hopefully you took into account that this particular task is intended to make restful api. That means we'll have separated methods for creating and updating single entities. So single save method is just useless here and much more complicated then we actually need. And usage of bulk operations is supposed to be avoided here, because we remove them from API contract.

Reply:
It is not clear for me what's the purpose of this sample. If it is just demonstrates true restful API creation - I agree. If it is an example of real-life extension module - I'd expect it to be consistent with other modules style.
Probably clear description of the task/PR could be useful to get the purpose of the module.

Reply:
@yecli , I do agree. We'll decide, formulate and provide the description.

Reply:
@yecli consistency makes sense only when tasks are similar. In this case task is different. While in other modules you are able to add/edit multiple entities, like catalog items, users, etc., although there is no need to being able to add or edit multiple customer reviews, because user can add only one review to catalog item at time, he can edit only one review, he can delete only one review. Why should I use "common" approach in this situation?

Reply:
@IvanSalov I do agree, it depends on the task. That's why I want to see description/purpose of this module.